jgrapht-core/src/main/java/org/jgrapht/alg/treedecomposition/IntervalgraphDecomposition.java

```diff
@@ -0,0 +1,347 @@
+        T extends Comparable<T>> IntervalgraphDecomposition<T, V> create(
+            IntervalGraph<V, E, VertexType, T> intervalGraph)
+    {
+        Set<V> vertexSet = intervalGraph.vertexSet();
+        List<Interval<T>> intervals = new ArrayList<Interval<T>>(vertexSet.size());
+        Map<Interval<T>, V> intervalToVertexMap = new HashMap<>(vertexSet.size());
+        Map<V, Interval<T>> vertexToIntervalMap = new HashMap<>(vertexSet.size());
+        for (V iv : vertexSet) {
+            intervals.add(iv.getInterval());
+            intervalToVertexMap.put(iv.getInterval(), iv);
+            vertexToIntervalMap.put(iv, iv.getInterval());
+        }
+        ArrayList<Interval<T>> startSort = new ArrayList<>(intervals);
+        ArrayList<Interval<T>> endSort = new ArrayList<>(intervals);
+        startSort.sort(Interval.<T> getStartingComparator());
+        endSort.sort(Interval.<T> getEndingComparator());
+        return new IntervalgraphDecomposition<T, V>(
+            startSort, endSort, intervalToVertexMap, vertexToIntervalMap);
+    }
+
+    /**
+     * Factory method for creating a nice tree decomposition for interval graphs. This factory
+     * method needs to lists of intervals, the first sorted after starting points, the second after
+     * ending points The complexity of this method is linear in the number of intervals
+     * 
+     * @param sortedByStartPoint a list of all intervals sorted by the starting point
+     * @param sortedByEndPoint a list of all intervals sorted by the ending point
+     * @param <T> the value of the intervals
+     * @return the algorithm for the computation of the nice tree decomposition
+     */
+    public static <T extends Comparable<T>> IntervalgraphDecomposition<T, Interval<T>> create(
+        List<Interval<T>> sortedByStartPoint, List<Interval<T>> sortedByEndPoint)
+    {
+        HashMap<Interval<T>, Interval<T>> identity = new HashMap<>(sortedByStartPoint.size());
+        for (Interval<T> interval : sortedByStartPoint)
+            identity.put(interval, interval);
+        return new IntervalgraphDecomposition<T, Interval<T>>(
+            new ArrayList<Interval<T>>(sortedByStartPoint),
+            new ArrayList<Interval<T>>(sortedByEndPoint), identity, identity);
+    }
+
+    /**
+     * Factory method for creating a nice tree decomposition for interval graphs. This factory
+     * method needs to lists of intervals, which then is sorted by ArrayList.sort(). The complexity
+     * of this method depends on the sorting Algorithm of ArrayList (probably O(|List| log(|List|))
+     * 
+     * @param intervals the (unsorted) list of all intervals
+     * @param <T> the values of the intervals
+     * @return the algorithm for the computation of the nice tree decomposition
+     * @see ArrayList#sort(Comparator)
+     */
+    public static <T extends Comparable<T>> IntervalgraphDecomposition<T, Interval<T>> create(
```

Comment:
is it possible to somehow refactor these three overloads in a way that one calls another to avoid duplicates?